### DISCUSSION TITLE
How to call function for interactive.bool?

### DISCUSSION THREAD

**Discussion by rrolla** [savonet/liquidsoap#2067] [User Support]
Hey, cannot understand how can i check if interactive is true and how to call some function if its truthy

OS: ubuntu:20.04 on docker
Liquidsoap version: https://github.com/savonet/liquidsoap/releases/tag/v2.0.0 installed from deb package

```ruby
#!/usr/bin/liquidsoap

interactive.harbor()
jingleOv = interactive.bool("jingleOver", false)

if jingleOv() then
  print("interactive on")
end

output.dummy(fallible=true, single("fallback.mp3"))
```

My goal is from web ui http://localhost:8000/interactive call function when its checked

![image](https://user-images.githubusercontent.com/3932546/142744921-c72587c7-cb58-4c6c-80ea-4dc010890cf4.png)

**Comment by smimram** [comment]
I would use a thread for this and do something like
```
interactive.harbor()
jingleOv = interactive.bool("jingleOver", false)

def check ()
  if jingleOv() then
    print("interactive on")
  end
end

thread.run(every=1., check)
```
which will poll for changes in the value every second.

**Reply by rrolla** [reply]
Hmm okay thanks will try, but how this code understands that it changed? there is no need to use thread.run

```ruby
# Register a telnet variable named volume with 1 as initial value
v = interactive.float("volume", 1.)

# Change the volume accordingly
source = amplify(v, source)
```

**Reply by smimram** [reply]
In oder to know whether it has changed you can compare with previous value. Or you can use
```
jingleOv = getter.changes(interactive.bool("jingleOver", false))
```
which will do this for you

**Reply by rrolla** [reply]
@smimram with thread `thread.run(every=1., check)` i got it working, but I imagine in real-time when I tick jingleOver then it plays immediate my jingle. And next question is after tick I set as false on liq script
```
jingleOv.set(false)
```
unfortunately, these changes are not reflected on the web UI, also after reload it still shows its ticked, but actually it's false
